lib/routes/routeBackbeat.js

```diff
@@ -1289,7 +1289,10 @@ function routeBackbeat(clientIP, request, response, log) {
                     [request.query.operation](request, response, log, next);
             }
             const versioningConfig = bucketInfo.getVersioningConfiguration();
-            if (!versioningConfig || versioningConfig.Status !== 'Enabled') {
+            // The following makes sure that only replication destination-related operations
+            // target buckets with versioning enabled.
+            const isVersioningEnabled = request.headers['x-scal-versioning-enabled'] === 'true';
```

Comment:
nit: `isVersioningEnabled` should be something like `checkVersioningEnabled`

similarly, i wonder if the header should be renamed to something more explicit, like `x-scal-versioning-required` or `x-scal-if-versioning-enabled` ?